frontend-guide.md

```diff
@@ -0,0 +1,333 @@
+Browserify is supplemented by [`babelify`](https://github.com/babel/babelify) which compile our ES6 code into browser compatible Javascript (learn more about [Babel](http://babeljs.io/]).
+
+#### Flow
+
+See Stack/Flow
+
+
+## Stack
+
+The stack of the frontend is almost the same for each project with few exceptions.
+
+### Backbone Collections and Models
+
+When calling an API resource, we use Backbone Collections and Models to transform them into usable, type-checked, javascript usuable class.
+Every resource is atomically defined by it's model. Each model contains a `parse` function validating and transforming data from API response.
+
+_Example of the model parse function_
+```javascript
+parse: function(response: Object): Object {
+  // Validating the API response contains the keys we expect and the values are the type we expect
+  const model = recordTS(
+    { id: numberTS
+    , name: stringTS
+    , createdAt: dateTS
+    , completedAt: nullableTS(dateTS)
+    , studentIds: arrayTS(numberTS)
+    }
+  )
+  const newResponse = objectOfType(response, model)
+  // Converting date from string to moment
+  newResponse.createdAt = moment(newResponse.createdAt)
+  newResponse.completedAt = newResponse.completedAt ? moment(newResponse.completedAt) : null
+  return newResponse
+}
+```
+
+To add more type safety around Backbone Model we add some boilerplate around the model creation.
+For each model we define the type of the return of the `get` function for each valid key. Calling `get` on a non valid `key` should throw a flow error, same for using the value as the incorrect type.
+To achieve this we have to define the type of the `Get` function as followed.
+From the same model as above
+
+```javascript
+type Get =
+  ((attr: 'id') => number) &
+  ((attr: 'name') => string) &
+  ((attr: 'createdAt') => moment) &
+  ((attr: 'completedAt') => ?moment) &
+  ((attr: 'studentIds') => Array<number>)
+```
+
+And attach the type of the `Get` function to the Backbone Model:
+
+```javascript
+export type MyModelT = Backbone.Model<Get> & typeof instanceMethods
+const MyModel: Class<Backbone.Model<Get>> = Backbone.Model.extend(instanceMethods)
+export default MyModel
+```
+
+More info about [Backbone])http://backbonejs.org/). You can also check the [annotated sources of Backbone](http://backbonejs.org/docs/backbone.html)
+
+
+### Backbone Router
+
+TODO
+
+### Flow
+
+[Flow](https://flow.org/) is a tool that add a layer of typing to javascript code. It relies on annotations in the code to figure out which type are the elements. When running `flow check`, flow will check the code base and throw all typing errors found. Flow contains the definition of the javascript API and can be completed by passing definition of libraries we use, like React, Backbone or Lodash. Other library definition can be in [flow-typed](https://github.com/flowtype/flow-typed).
```

Comment:
I would say "It relies on annotations in the code to determine the types of expressions". Flow also does some type inference, but it's really weak, and only improves if you have (usually) function-argument annotations.